pkg/services/ngalert/notifier/channels/template_data.go

```diff
@@ -149,6 +149,7 @@ func TmplText(ctx context.Context, tmpl *template.Template, alerts []*types.Aler
 	data := ExtendData(promTmplData, l)
 
 	return func(name string) (s string) {
+		// TODO: Make templating function return new error each time instead of passing pointer by parameter
```

Comment:
This is the way that Prometheus handles errors. It was introduced in [this PR](https://github.com/prometheus/alertmanager/pull/1474).
It's kind of an ongoing discussion on [Slack](https://raintank-corp.slack.com/archives/C028MCV4R7C/p1663706834956509). We should come back to this at some point.